[BUG] Out of bounds access in IO Uncomp

**Describe the bug**
There is a out-of-bounds array access warning when compiling `io/comp/uncomp.cpp`:

```bash
[ 27%] Building CXX object CMakeFiles/cudf.dir/src/io/comp/uncomp.cpp.o
/home/jieong/scratch/cudf/cpp/src/io/comp/uncomp.cpp:367:86: warning: array index 3 is past the end of the array
      (which contains 3 elements) [-Warray-bounds]
            if (fhdr->sig[0] == 'B' && fhdr->sig[1] == 'Z' && fhdr->sig[2] == 'h' && fhdr->sig[3] >= '1' && fhdr->sig[3] <= '9')
                                                                                     ^         ~
/home/jieong/scratch/cudf/cpp/src/io/comp/uncomp.cpp:102:5: note: array 'sig' declared here
    uint8_t sig[3]; // "BZh"
    ^
/home/jieong/scratch/cudf/cpp/src/io/comp/uncomp.cpp:367:109: warning: array index 3 is past the end of the array
      (which contains 3 elements) [-Warray-bounds]
            if (fhdr->sig[0] == 'B' && fhdr->sig[1] == 'Z' && fhdr->sig[2] == 'h' && fhdr->sig[3] >= '1' && fhdr->sig[3] <= '9')
                                                                                                            ^         ~
/home/jieong/scratch/cudf/cpp/src/io/comp/uncomp.cpp:102:5: note: array 'sig' declared here
    uint8_t sig[3]; // "BZh"
    ^
2 warnings generated.
```

**Steps/Code to reproduce bug**
1. Enable stricter warning checks in GCC or Clang
2. Build cuDF using the build instructions

**Expected behavior**
Fix the out-of-bounds array warning.

**Environment details (please complete the following information):**
 - Environment location: Bare-metal
 - Method of cuDF install: conda

**Additional context**
Error is not encountered using the default compiler build settings.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -23,6 +23,7 @@
 - PR #1254 CSV Reader: fix data type detection for floating-point numbers in scientific notation
 - PR #1289 Fix looping over each value instead of each category in concatenation
 - PR #1293 Fix Inaccurate error message in join.pyx
+- PR #1321 Fix out of bounds warning when checking Bzip2 header
 
 
 # cuDF 0.6.0 (Date TBD)

--- a/cpp/src/io/comp/uncomp.cpp
+++ b/cpp/src/io/comp/uncomp.cpp
@@ -364,7 +364,8 @@ gdf_error io_uncompress_single_h2d(const void *src, gdf_size_type src_size, int 
         {
             const bz2_file_header_s *fhdr = (const bz2_file_header_s *)raw;
             // Check for BZIP2 file signature "BZh1" to "BZh9"
-            if (fhdr->sig[0] == 'B' && fhdr->sig[1] == 'Z' && fhdr->sig[2] == 'h' && fhdr->sig[3] >= '1' && fhdr->sig[3] <= '9')
+            if (fhdr->sig[0] == 'B' && fhdr->sig[1] == 'Z' && fhdr->sig[2] == 'h' &&
+                fhdr->blksz >= '1' && fhdr->blksz <= '9')
             {
                 strm_type = IO_UNCOMP_STREAM_TYPE_BZIP2;
                 comp_data = raw;
@@ -373,7 +374,7 @@ gdf_error io_uncompress_single_h2d(const void *src, gdf_size_type src_size, int 
             }
         }
         if (strm_type != IO_UNCOMP_STREAM_TYPE_INFER)
-            break; // Fall through for INFER     
+            break; // Fall through for INFER
     default:
         // Unsupported format
         break;

